### PR TITLE
Syntax Lookup: More coercion docs

### PR DESCRIPTION
**Pull request by fhammerschmidt**

### CODE REVIEW

File: misc_docs/syntax/operators_type_coercion.mdx [hunk 1]
```diff
@@ -8,11 +8,20 @@ category: "operators"
 
 The `:>` operator may be used to convert a polymorphic variant to a `string` or `int`, or convert an [object](/docs/manual/latest/object) to a type with a subset of its fields.
 
+_Since ReScript `11.0.0`_ coercion also works for converting
+- from `int` to `float`
+- from record to record with the same field(s)
```
Comment (zth) [inline misc_docs/syntax/operators_type_coercion.mdx:13]:
```suggestion
- from record to record with the same field(s), or when record A is a subtype of record B
```

File: misc_docs/syntax/operators_type_coercion.mdx [hunk 2]
```diff
@@ -8,11 +8,20 @@ category: "operators"
...
+- from `@unboxed` variant with only strings to `string`
+- from `string` to `@unboxed` variant that have a catch-all unboxed `string` case
+- from variant to variant when applicable
+- from variant to `string`/`int`/`float` when applicable
+- for invariant type arguments such as array payloads
```
Comment (zth) [inline misc_docs/syntax/operators_type_coercion.mdx:18]:
```suggestion
- for invariant type arguments such as array payloads when the runtime representation is guaranteed to be exactly the same
```

**Review by cristianoc — APPROVED** [review]
nice one

**Comment by fhammerschmidt** [comment]
Can't fix tests for now since I would need to install the unreleased rc.8 for it.

**Pull request merged**